### PR TITLE
Create a new page for Argo CD resource requirements

### PR DESCRIPTION
**Pull request by iam-veeramalla**
Signed-off-by: iam-veeramalla <abhishek.veeramalla@gmail.com>

**What type of PR is this?**
> /kind documentation

**What does this PR do / why we need it**:
This PR creates a page to explain the creation, updation and deletion of resource requests/limits for Argo CD workloads.

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**How to test changes / Special notes to the reviewer**:
Install mkdocs
Run `mkdocs serve` to start the UI locally on `0.0.0.0:8000`
A new page can be seen with name `Resource Requirements`
Screenshot 2021-08-31 at 1 15 14 AM

### CODE REVIEW

File: docs/usage/resource_management.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Resource Requirements
+
+This page covers the steps to create, update and delete resource requests and limits for Argo CD workloads.
+
+The Argo CD Custom Resource allows you to create the workloads with desired resource requests and limits. This is required when a user/admin wish to deploy his Argo CD instance in a namespace that is set with [Resource Quota](https://kubernetes.io/docs/concepts/policy/resource-quotas/).
```
Comment (tylerauerbeck) [inline docs/usage/resource_management.md:5]: @iam-veeramalla Could this be "This is required when a user/admin **wishes** to deploy **their** ArgoCD instance ....."

**Review by tylerauerbeck — COMMENTED** [review]
@iam-veeramalla Just a small comment. Otherwise LGTM.

File: docs/usage/resource_management.md (hunk 1 above)
Comment (iam-veeramalla) [reply docs/usage/resource_management.md]: Thanks @tylerauerbeck. Corrected it :)

**Review by jannfis — APPROVED** [review]
LGTM

**Pull request merged**